Mathlib/Analysis/SpecialFunctions/Pow/NNReal.lean

```diff
@@ -561,12 +561,24 @@ theorem rpow_eq_top_of_nonneg (x : ℝ≥0∞) {y : ℝ} (hy0 : 0 ≤ y) : x ^ y
     exact h.right hy0
   · exact h.left
 
+-- This is an unsafe rule since we want to try `rpow_pos` if x = 0.
+@[aesop (rule_sets := [finiteness]) unsafe apply]
 theorem rpow_ne_top_of_nonneg {x : ℝ≥0∞} {y : ℝ} (hy0 : 0 ≤ y) (h : x ≠ ⊤) : x ^ y ≠ ⊤ :=
   mt (ENNReal.rpow_eq_top_of_nonneg x hy0) h
 
 theorem rpow_lt_top_of_nonneg {x : ℝ≥0∞} {y : ℝ} (hy0 : 0 ≤ y) (h : x ≠ ⊤) : x ^ y < ⊤ :=
   lt_top_iff_ne_top.mpr (ENNReal.rpow_ne_top_of_nonneg hy0 h)
 
+lemma rpow_ne_top {x : ℝ≥0∞} {y : ℝ} : ¬(x = 0 ∧ y < 0 ∨ x = ⊤ ∧ 0 < y) → x ^ y ≠ ⊤ := by
```

Comment:
That's a weird spelling (which is handy for the proof, I understand). Wouldn't it be more natural to have two assumptions, of the form `x ne 0 or y nonneg` and `x ne top or y nonpos`?

Reply:
Good question! I changed this to a version with two assumptions (which seems to be longer, though --- perhaps I'm missing a golfing trick).

Reply:
It turns out the new lemmas is in fact not needed: I have deleted it now.